### PR TITLE
Fixing diagnostic on metadata failure

### PR DESCRIPTION
**Pull request by simplynaveen20**
During DR drill we have noticed on any meta data failure connection mode is showing Gateway instead of Direct .
Also there was no indication which metadata failed. Fixing these two issues in this PR

Below added new resource type in gateway statistics and moving connection mode to client config -

image

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ClientSideRequestStatistics.java [hunk 1]
```diff
@@ -129,7 +127,12 @@ public void recordGatewayResponse(
             }
             this.gatewayStatistics = new GatewayStatistics();
             if (rxDocumentServiceRequest != null) {
+                if (rxDocumentServiceRequest.getResourceType().equals(ResourceType.Document)) {
```
Comment (j82w) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ClientSideRequestStatistics.java:130]: Can you include a before and after in the PR description to show the changes to the JSON output?

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ClientSideRequestStatistics.java [hunk 2]
```diff
@@ -129,7 +127,12 @@ public void recordGatewayResponse(
...
+                    connectionMode = ConnectionMode.GATEWAY;
```
Comment (j82w) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ClientSideRequestStatistics.java:131]: Doesn't the client configuration have this information? Why is it needed on the ClientSideRequestStatistics?

Comment (simplynaveen20) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ClientSideRequestStatistics.java:131]: Currently client configuration doesnt have connection mode ,  i was trying to make scope of this PR small, but let change in client configuration in this PR itself

**Review by kushagraThapar — APPROVED** [review]
LGTM

**Review by j82w — APPROVED** [review]

**Pull request merged**